### PR TITLE
Sort module documentation pages

### PR DESCRIPTION
**Pull request by albi3ro**
As PennyLane grows, so does the number of modules and therefore the number of module documentation pages.  This PR explores better ways to sort the module documentation pages on the website.

It breaks the modules into the general "API" modules and "Internal API".

Each category is sorted alphabetically.

### CODE REVIEW

**Review by josh146 — APPROVED** [review]
Looks good to me @albi3ro! Only a minor comment - the phrasing 'Internal API' gives the impression that these are actually APIs. However, from looking at the TOC, it is really things a PL core developer will need to know.

Is API the right phrasing here? It implies a maintained contract which may not be the case?

I have to admit I cannot think of better phrasing.

**Review by antalszava — APPROVED** [review]
Looks good :+1:

**Comment by antalszava** [comment]
As per Josh's point: what if we replaced "API" with "Modules"? These are modules that we're documenting, simply having "Modules" and "Internal modules" makes sense to me personally.

File: doc/index.rst [hunk 1]
```diff
@@ -173,24 +173,30 @@ PennyLane is **free** and **open source**, released under the Apache License, Ve
...
+   code/qml_transforms
    code/qml_qaoa
    code/qml_qchem
    code/qml_qnn
-   code/qml_resource
-   code/qml_shadows
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Internal API
```
Comment (albi3ro) [inline doc/index.rst:194]:
```suggestion
   :caption: Internals
```

**Comment by albi3ro** [comment]
@antalszava @josh146 I just renamed "Internal API" to just "Internals".

**Comment by josh146** [comment]
Nice :+1:

**Pull request merged**